lib/fortnox/api/types/nullable.rb

```diff
@@ -8,13 +8,17 @@ module Types
       module Nullable
         String = Types::Strict::String.optional.constructor { |value| value&.to_s }
         Float = Types::Strict::Float.optional.constructor { |value| value&.to_f }
-        Integer = Types::Strict::Int.optional.constructor { |value| value&.to_i }
-        Boolean = Types::Bool.optional.constructor { |value| THE_TRUTH.fetch(value) { false } }
-        # TODO: Improve parsing!
-        # In case date parsing fails, ArgumentError is thrown. Currently, it is rescued in Repository::Loaders.find.
-        # That method assumes that the exception is due to invalid argument to the find method, which is not the case
-        # if it is raised from here!
-        Date = Types::Date.optional.constructor { |value| ::Date.parse(value.to_s) unless value.nil? || value == '' }
+        Integer = Types::Strict::Integer.optional.constructor { |value| value&.to_i }
+        Boolean = Types::Bool.optional.constructor { |value| THE_TRUTH.fetch(value, false) }
+        Date = Types::Date.optional.constructor do |value|
+          next if value.nil? || value == ''
+
+          begin
+            ::Date.parse(value.to_s)
+          rescue StandardError
```

Comment:
I'm thinking about this change, do we really want to catch the error here and return `nil`? What if we provide an invalid date to our model, don't we want to raise an `Fortnox::API::AttributeError` then, as we do in `Fortnox::API::Model::Base`? 🤔 I think that's where we catch it today, rather than in the `Repository::Loaders.find` that the comment says - there's no `rescue` statement in that class...